flask_monitoringdashboard/database/request.py

```diff
@@ -8,33 +8,31 @@
 from flask_monitoringdashboard.database import Request
 
 
-def get_latencies_in_timeframe(db_session, endpoint_id, start_date, end_date):
-    criterion = create_time_based_sample_criterion(start_date, end_date)
-
-    items = db_session.query(Request.duration).filter(Request.endpoint_id == endpoint_id, *criterion).all()
+def get_latencies_in_timeframe(db_session, endpoint_id, criterion):
+    items = db_session.query(Request.duration).filter(Request.endpoint_id == endpoint_id,
+                                                      *criterion).all()
 
     return [item.duration for item in items]
 
 
-def get_latencies_sample(db_session, endpoint_id, interval, sample_size=500):
-    criterion = create_time_based_sample_criterion(interval.start_date(), interval.end_date())
-
+def get_latencies_sample(db_session, endpoint_id, criterion, sample_size=500):
     dialect = db_session.bind.dialect.name
 
     if dialect in ['sqlite', 'mysql']:
         order_by = func.random() if dialect == 'sqlite' else func.rand()
 
         items = db_session.query(Request.duration) \
-            .filter(Request.endpoint_id == endpoint_id, *criterion) \
+            .filter(Request.endpoint_id == endpoint_id, criterion) \
```

Comment:
shouldn't you have *criterion here? 
or alternatively, above in get_latencies_in_timeframe not have a star?

Reply:
Honestly, not sure yet. The asterisks are confusing me today 🙄

It seems to be working now though ...

Reply:
The * unpacks iterable objects. There's an inconsistency now, because criterion in line 25 has to be a single criterion, while the same argument is passed to the function above, which expects a list of criterions (criteria?) in line 13.

Reply:
@thejager - any updates on this?